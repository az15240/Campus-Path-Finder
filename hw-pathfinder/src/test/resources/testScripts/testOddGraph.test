# Test an odd graph in which there's a center node connecting to every other nodes.
# Every "other nodes" connect to two other "other nodes".
# The shape of graph is like a symmetrical polygon.

CreateGraph A
AddNode A n0
AddNode A n1
AddNode A n2
AddNode A n3
AddNode A n4
AddNode A n5
AddEdge A n0 n1 0.1
AddEdge A n0 n2 0.2
AddEdge A n0 n3 0.3
AddEdge A n0 n4 0.4
AddEdge A n0 n5 0.5
AddEdge A n1 n2 1.2
AddEdge A n1 n5 1.5
AddEdge A n2 n1 2.1
AddEdge A n2 n3 2.3
AddEdge A n3 n2 3.2
AddEdge A n3 n4 3.4
AddEdge A n4 n3 4.3
AddEdge A n4 n5 4.5
AddEdge A n5 n1 5.1
AddEdge A n5 n4 5.4
ListNodes A
ListChildren A n0
ListChildren A n1
ListChildren A n2
ListChildren A n3
ListChildren A n4
ListChildren A n5

FindPath A n1 n3
FindPath A n2 n5
FindPath A n3 n5