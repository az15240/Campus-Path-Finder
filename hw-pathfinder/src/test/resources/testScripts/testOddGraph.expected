# Test an odd graph in which there's a center node connecting to every other nodes.
# Every "other nodes" connect to two other "other nodes".
# The shape of graph is like a symmetrical polygon.

created graph A
added node n0 to A
added node n1 to A
added node n2 to A
added node n3 to A
added node n4 to A
added node n5 to A
added edge 0.100 from n0 to n1 in A
added edge 0.200 from n0 to n2 in A
added edge 0.300 from n0 to n3 in A
added edge 0.400 from n0 to n4 in A
added edge 0.500 from n0 to n5 in A
added edge 1.200 from n1 to n2 in A
added edge 1.500 from n1 to n5 in A
added edge 2.100 from n2 to n1 in A
added edge 2.300 from n2 to n3 in A
added edge 3.200 from n3 to n2 in A
added edge 3.400 from n3 to n4 in A
added edge 4.300 from n4 to n3 in A
added edge 4.500 from n4 to n5 in A
added edge 5.100 from n5 to n1 in A
added edge 5.400 from n5 to n4 in A
A contains: n0 n1 n2 n3 n4 n5
the children of n0 in A are: n1(0.100) n2(0.200) n3(0.300) n4(0.400) n5(0.500)
the children of n1 in A are: n2(1.200) n5(1.500)
the children of n2 in A are: n1(2.100) n3(2.300)
the children of n3 in A are: n2(3.200) n4(3.400)
the children of n4 in A are: n3(4.300) n5(4.500)
the children of n5 in A are: n1(5.100) n4(5.400)

path from n1 to n3:
n1 to n2 with weight 1.200
n2 to n3 with weight 2.300
total cost: 3.500
path from n2 to n5:
n2 to n1 with weight 2.100
n1 to n5 with weight 1.500
total cost: 3.600
path from n3 to n5:
n3 to n2 with weight 3.200
n2 to n1 with weight 2.100
n1 to n5 with weight 1.500
total cost: 6.800